numpy/lib/histograms.py

```diff
@@ -605,13 +605,14 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
         (instead of 1). If `density` is True, the weights are
         normalized, so that the integral of the density over the range
         remains 1.
-    density : bool, optional
+    density : bool or str, optional
         If ``False``, the result will contain the number of samples in
         each bin. If ``True``, the result is the value of the
         probability *density* function at the bin, normalized such that
-        the *integral* over the range is 1. Note that the sum of the
-        histogram values will not be equal to 1 unless bins of unity
-        width are chosen; it is not a probability *mass* function.
+        the *integral* over the range is 1. 
+        If ``sum``, the result will be normalized such that the sum of 
+        the histogram values will be equal to 1. This *is* a probability 
+        mass function. 
```

Comment:
It seems very peculiar to me to use the `density` argument to request a `mass` function. I'd prefer a `mass=True` argument to acheive that, or perhaps `mode='count'/'density'/'mass'`.

I think there have been discussions about this before, either here or on the mailing list.